src/app.rs

```diff
@@ -0,0 +1,103 @@
+use std::net::SocketAddr;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::net::{TcpListener, ToSocketAddrs};
+
+use crate::router::{Endpoint, Router};
+use crate::server::serve;
+use crate::{Error, Middleware};
+
+/// The default value of the maximum number of concurrent connections.
+const DEFAULT_MAX_CONNECTIONS: usize = 256;
+
+/// The default value of the shutdown timeout in seconds.
+const DEFAULT_SHUTDOWN_TIMEOUT: u32 = 30;
+
+pub struct App<State> {
```

Comment:
I'm someone with a preference for single letter generic type parameters. However, I noticed while working on this project how beneficial it is when opening a file at a glance, being able to recognize that a type or function takes `State` as a generic parameter. It makes it very easy to how state moves throughout an application with no ambiguity. Sustainability is important to me as I want to be able to dedicate some of time to this long term. For that reason, I'll consider this first and perhaps only exception to my preference of single letter generic type parameters.